tests/extended_temporal_memory/extensive_etm_test.py

```diff
@@ -0,0 +1,1274 @@
+    self.assertGreaterEqual(unpredictedActiveColumns, min(self.w))
+
+
+  def testE11(self):
+    """Repeated motor command copy as external input."""
+    self.init()
+
+    numbers = self.sequenceMachine.generateNumbers(1, 100)
+    proximalInputA = self.sequenceMachine.generateFromNumbers(numbers)
+
+    numbers = [100, 200] * 50 + [None]
+    externalInputA = self.sequenceMachine.generateFromNumbers(numbers)
+
+
+    for _ in xrange(2):
+      self.feedTM(proximalInputA, activeExternalCellsSequence=externalInputA)
+
+    self._testTM(proximalInputA, activeExternalCellsSequence=externalInputA)
+    self.assertAllActiveWerePredicted()
+    self.assertAllInactiveWereUnpredicted()
+
+
+  def testE12(self):
+    """Simple learnOnOneCell test.
+
+    Train on ABCADC. Without learnOnOneCell, C should have different representations in ABC and ADC.
+    """
+    self.init({"learnOnOneCell": False})
+    self.assertFalse(self.tm.learnOnOneCell)
+
+    numbers = self.sequenceMachine.generateNumbers(1, 10)
+    numbers[0] = 50
+    numbers[-2] = 75
+    proximalInputA = self.sequenceMachine.generateFromNumbers(numbers)
+
+    numbers = self.sequenceMachine.generateNumbers(1, 10)
+    numbers[0] = 50
+    numbers[-2] = 75
+    proximalInputB = self.sequenceMachine.generateFromNumbers(numbers)
+
+    trainInput = proximalInputA[:-1] + proximalInputB
+
+    for _ in xrange(2):
+      self.feedTM(trainInput)
+
+    self._testTM(proximalInputA)
+    predictedActiveA = self.tm.mmGetTracePredictedActiveCells().data[-1]
+    self._testTM(proximalInputB)
+    predictedActiveB = self.tm.mmGetTracePredictedActiveCells().data[-1]
+    # check that the two representations are the same
+    self.assertNotEquals(predictedActiveA, predictedActiveB)
+
+
+  def testE13(self):
+    """Simple learnOnOneCell test.
+
+    Train on ABCADC, check that C has the same representation in ADC and ABC.
+    """
+    self.init({"learnOnOneCell": True})
+    self.assertTrue(self.tm.learnOnOneCell)
+
+    numbers = self.sequenceMachine.generateNumbers(1, 10)
+    numbers[0] = 50
+    numbers[-2] = 75
+    proximalInputA = self.sequenceMachine.generateFromNumbers(numbers)
+
+    numbers = self.sequenceMachine.generateNumbers(1, 10)
+    numbers[0] = 50
+    numbers[-2] = 75
+    proximalInputB = self.sequenceMachine.generateFromNumbers(numbers)
+
+    trainInput = proximalInputA[:-1] + proximalInputB
+
+    for _ in xrange(2):
+      self.feedTM(trainInput)
+
+    self._testTM(proximalInputA)
+    predictedActiveA = self.tm.mmGetTracePredictedActiveCells().data[-1]
+    self._testTM(proximalInputB)
+    predictedActiveB = self.tm.mmGetTracePredictedActiveCells().data[-1]
+    # check that the two representations are the same
+    self.assertEqual(predictedActiveA, predictedActiveB)
+
+
+  def testE14(self):
+    """learnOnOneCell with motor command should not impair behavior.
+
+    Using learnOnOneCell, does the same test as E6, using the same parameters.
+    """
```

Comment:
This is the test that fails. Whereas we expect the ETM to learn this simple pattern, a few columns (0 to 5) burst at each step. This is because of the winning cell mechanism, which has a slight flaw when using learnOnOneCell: since each synapse needs two "increments" to become connected, and the least used cell is chosen, for each column in the first pattern of the sequence, all cells will have to be selected once before going back to the first one and connecting the synapse. The consequence of this is that the sequence has to be fed exactly (#columns + 1) times for the test to pass.